### PR TITLE
OJ-33816: fix bad logging call in agent

### PR DESCRIPTION
**Pull request by gavinpitt-jf**
## Description

There is a bad logging call deep in the BBCloud rate limiter logic that rarely gets hit. One of our clients, however, has hit it and failed their git agent upload on it.

### CODE REVIEW

**Review by thingscouldbeworse — APPROVED** [review]

File: jf_agent/jf_jira/jira_download.py [hunk 1]
```diff
@@ -41,9 +41,7 @@ def download_users(
     # searching a different way
     if 950 <= len(jira_users) <= 1000:
         logger.info(
-            level=logging.INFO,
-            msg=f'Page limit reached with {len(jira_users)} users, '
-            'falling back to search by letter method.',
+            f'Page limit reached with {len(jira_users)} users, falling back to search by letter method.'
```
Comment (gavinpitt-jf) [inline jf_agent/jf_jira/jira_download.py:44]: This was the same bug that I found in `ratelimit.py`, but somewhere else. We haven't hit here yet, thankfully, but this shouldn't error our like it did for the ratelimit

File: jf_agent/jf_jira/jira_download.py [hunk 2]
```diff
@@ -494,7 +492,7 @@ def detect_issues_needing_re_download(
         existing_metadata = issue_metadata_from_jellyfish.get(int(issue_id_str))
         if existing_metadata and issue_key != existing_metadata.key:
             logger.info(
-                f'Detected a key change for issue {issue_id_str} ({existing_metadata.key} -> {issue_key})',
+                f'Detected a key change for issue {issue_id_str} ({existing_metadata.key} -> {issue_key})'
```
Comment (gavinpitt-jf) [inline jf_agent/jf_jira/jira_download.py:495]: Black formatting

File: jf_agent/jf_jira/jira_download.py [hunk 3]
```diff
@@ -672,7 +670,7 @@ def _download_jira_issues_segment(
 
 
 def _download_jira_issues_page(
-        jira_connection, jira_issue_ids_segment, field_spec, start_at, batch_size
+    jira_connection, jira_issue_ids_segment, field_spec, start_at, batch_size
```
Comment (gavinpitt-jf) [inline jf_agent/jf_jira/jira_download.py:673]: Black formatting

File: jf_agent/jf_jira/jira_download.py [hunk 4]
```diff
@@ -695,7 +693,8 @@ def _download_jira_issues_page(
             resp_json = json_loads(
                 retry_for_429s(
                     jira_connection._session.post,
-                        url=jira_connection._get_url('search'), data=json.dumps(search_params)
+                    url=jira_connection._get_url('search'),
+                    data=json.dumps(search_params),
```
Comment (gavinpitt-jf) [inline jf_agent/jf_jira/jira_download.py:697]: black formatting

File: jf_agent/ratelimit.py [hunk 5]
```diff
@@ -70,8 +70,7 @@ def limit(self, realm):
             sleep_period_secs = (sleep_until - datetime.utcnow()).total_seconds()
             if sleep_period_secs > 0:  # it's possible that sleep_until was a couple ms ago
                 logger.info(
-                    logging.INFO,
-                    f'Sleeping for {sleep_period_secs:.1f} secs ({sleep_period_secs / 60.0:.1f} mins)',
+                    f'Sleeping for {sleep_period_secs:.1f} secs ({sleep_period_secs / 60.0:.1f} mins)'
```
Comment (gavinpitt-jf) [inline jf_agent/ratelimit.py:73]: This was the 'root' bug

`logger.info` DOES NOT accept a `logging.INFO` call. This logger.info call must have previously been `logger.log`, which WOULD take the `logging.INFO` argument as the logging level. When we changed this field from `logger.log` to `logger.info` we forgot to remove the `logging.INFO` arg

**Pull request merged**